soc/arc/snps_arc_iot/soc_irq.h

```diff
@@ -1,117 +0,0 @@
-/*
- * Copyright (c) 2018 Synopsys, Inc. All rights reserved.
- *
- * SPDX-License-Identifier: Apache-2.0
- */
-
-#ifndef _ARC_IOT_SOC_IRQ_H_
-#define _ARC_IOT_SOC_IRQ_H_
-
-
-#define IRQ_WATCHDOG				18
```

Comment:
This file had as a convenience all interrupts defined. In Zephyr, not all devices/interrupts are supported, so it makes sense only the interrupts used where added to the dts files.
That will have to be extended if more devices are added/supported in the future.

Reply:
if more devices are added, interrupts should be defined in Devicetree (together with their priority), not in C files anymore. Zephyr coding guidelines state that there should not be any dead code, ref. https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html. Many ARM SoC HALs also define all interrupt numbers, but in Zephyr they're not used (and if they do it should be fixed).